sklearn/preprocessing/_encoders.py

```diff
@@ -158,6 +160,19 @@ class OneHotEncoder(_BaseEncoder):
 
         The used categories can be found in the ``categories_`` attribute.
 
+    drop : 'first' or a list/array of shape (n_features,), default=None.
+        Specifies a methodology to use to drop one of the categories per
+        feature. This is useful in situations where perfectly collinear
+        features cause problems, such as when feeding the resulting data
+        into a neural network or an unregularized regression.
+
+        - None : retain all features.
+        - 'first' : drop the first category in each feature. If only one
+                    category is present, the feature will be dropped
+                    entirely.
+        - array : ``drop[i]`` is the caetgory in ``X[:,i]`` that should be
+                  dropped. None means all categories should be retained.
```

Comment:
maybe add here something like "for that feature", to make it clear this can be specified per feature (and that it is different from the None option above)